### DISCUSSION TITLE
Mydumper/myloader issue with MySQL 8.0 version

### DISCUSSION THREAD

**Discussion by Patel1S** [mydumper/mydumper#1390] [Q&A]
Hi Everyone,

Currently I'm using MyDumper v0.15.1-3 built against MySQL 5.7.42-46 with SSL Support. And i'm facing issues whenever i'm trying to do backup restore from MySQL 5.7 database to MySQL 8.0 database.

Below are the errors we received during MyLoader script execution while doing the test:

1) **Error: gzip: stdout: Broken pipe**

- We are still receiving this error after we tried to use myloader command with one thread before we were using 4 threads. We are not facing this issue on MySQL 5.7 version. Only when we try to restore to MySQL 8.0 version database from MySQL 5.7. Also looked into all transaction timeout setting which is same for both MySQL 5.7 and 8.0 version cluster. And also I tried to use mysqldump backup/restore method from MySQL 5.7 database to MySQL 8.0 database that works fine. 

2) **Deadlock found when trying to get lock; try restarting transaction** - Resolved

-  By changing myloader command with one thread we didn't see this error again. Before we were using 4 threads.

Currently I have below questions:

1. Do we have any mydumper version release that is built against MySQL 8.0?
2. If we don't have mydumper release that is built against MySQL 8.0 version then do we have a way to build mydumper v0.15.1-3 agianst MySQL 8.0?  (I'm thinking to remove mydumper and MySQL 5.7 libraries and then reinstall MySQL 8.0 libraries and install mydumper but not sure how i can build against MySQL 8.0 version libraries?)

**Current System details:**
OS version: Red Hat Enterprise Linux (RHEL) 7
Mydumper version: mydumper v0.15.1-3, built against MySQL 5.7.42-46 with SSL support

**Comment by davidducos** [comment]
Hi @Patel1S 
Can you test latest prerelease?

**Reply by Patel1S** [reply]
Hi @davidducos i have installed new pre-release version [v0.15.2-8]. And tried doing test again with that but still its failing on myloader(restore) with below error,

**Error: ** (myloader:23675): CRITICAL **: 18:53:37.718: Drop view failed: 'schema.tablename' is not VIEW**

though its a table not sure why its taking it as view.

**Commands we're using:**
mydumper --host $hostsource --user $MYSQL_USER --password $gettokensource --port xxxx --compress-protocol --database ${db} --lock-all-tables --long-query-retries 20 --long-query-retry-interval 120 --long-query-guard 999999 --routines --views-as-tables --no-views --insert-ignore --compress --outputdir ${BACKUP_PATH}/ --dirty --logfile mydumper.log --threads 2 --verbose 3

**(I tried running mydumper command without --views-as-tables but had same error so i added it to command to see how it reacts but it's getting same above error)**


myloader --host $hostdestination --user $MYSQL_USER --password $gettokendestination --port xxxx --compress-protocol --skip-triggers --innodb-optimize-keys --overwrite-tables --retry-count 20 --directory ${BACKUP_PATH}/ --logfile myloader.log --threads 4 --verbose 3

**Comment by jbalatero** [comment]
I'm using [v0.16.3-2](https://hub.docker.com/layers/mydumper/mydumper/v0.16.3-2/images/sha256-a984500f3d4ef26d2150772067c7f0a28ac2247c1bbdaf10b733732f49a8b693?context=explore), but I'm still encountering this error.

**Comment by davidducos** [comment]
Hello @jbalatero/ @Patel1S , 
Since MySQL 8.0 we need to use --serialized-table-creation to avoid deadlock during table creation. 
Please, try that and let me know.

**Reply by jbalatero** [reply]
`--serialized-table-creation` worked! No more deadlock error

**Comment by Patel1S** [comment]
Sorry forgot to respond back but after applying v0.16.3-2 and making changes in loader command such as adding --serialized-table-creation. it's working fine since a while.
